File: js/src/lib/stack.js
export default class Stack {
    constructor(array) {
        this.array = [];
        if (array) {
            for (let i = 0; i < array.length; i++) {
                this.array.push(array[i]);
            }
        }
    }

    push(item) {
        this.array.push(item);
    }

    pop() {
        if (this.array.length > 0) {
            return this.array.pop();
        }
        throw "Stack is empty";
    }

    peek() {
        if (this.array.length > 0) {
            return this.array[this.array.length - 1];
        }
        throw "Stack is empty";
    }

    any() {
        return this.array.length > 0;
    }

    count() {
        return this.array.length;
    }

    toArray() {
        const result = [];
        for (let i = 0; i < this.array.length; i++)
            result[i] = this.array[this.array.length - i - 1];
        return result;
    }
}

File: python/lib/stack.py
from typing import Optional, Any, List


class StackIsEmptyException(Exception):
    pass


class Stack:

    def __init__(self, values: Optional[List[Any]] = None) -> None:
        self.stack = values or []

    def push(self, value: Any) -> None:
        self.stack.append(value)

    def pop(self) -> Any:
        if len(self.stack) <= 0:
            raise StackIsEmptyException
        return self.stack.pop()

    def top(self) -> List[Any]:
        if len(self.stack) <= 0:
            raise StackIsEmptyException
        return self.stack[-1]

    def to_list(self) -> List[Any]:
        return list(reversed(self.stack))

    def __len__(self) -> int:
        return len(self.stack)
